components/studentPortal/profileV2/achievement_components/CodingBadgeUnit.tsx

```diff
@@ -41,14 +43,21 @@ function CodingBadgeUnit({
   };
 
   return (
-    <div className="flex flex-col items-center justify-center h-full p-4 rounded-xl text-textPrimary bg-backgroundSecondary border-slate-300">
+    <div
+      className="flex flex-col items-center justify-center h-full p-4 text-white bg-slate-600 border-slate-300"
+      onClick={() => {
+        if (disabled) {
```

Comment:
this is not great. You're using the disabled prop to enable a link. That's confusing. Wonder if we should introduce a new property to control this link